cscs-checks/prgenv/src/alloc_speed.cpp

```diff
@@ -0,0 +1,27 @@
+#include <chrono>
+#include <iostream>
+#include <algorithm>
+
+double test_alloc(size_t n)
+{
+    auto t0 = std::chrono::system_clock::now();
+    /* time to allocate + fill */
+    char* ptr = (char*)std::malloc(n);
+    std::fill(ptr, ptr + n, 0);
+    auto t1 = std::chrono::system_clock::now();
+    /* time fo fill */
+    std::fill(ptr, ptr + n, 0);
+    auto t2 = std::chrono::system_clock::now();
+    t0 += static_cast<std::chrono::seconds>(ptr[0]);
+    std::free(ptr);
+
+    return std::chrono::duration_cast<std::chrono::duration<double>>((t1 - t0) - (t2 - t1)).count();
+}
+
+int main(int argn, char** argv)
+{
+    for (int i = 20; i < 33; i++) {
+        size_t sz = size_t(1) << i;
+        std::cout << sz / 1024.0 / 1024.0 << " MB, allocation time " << test_alloc(sz) << " sec. \n";
+    }
```

Comment:
Not really. The C++ [Standard states](https://github.com/cplusplus/draft/blob/5aaec13ef8f29527e762f8e4bf8e2362c8b1419c/source/basic.tex#L5774):

```
A \tcode{return} statement\iref{stmt.return} in \tcode{main} has the effect of leaving the main
function (destroying any objects with automatic storage duration) and
calling \tcode{std::exit} with the return value as the argument.
If control flows off the end of
the \grammarterm{compound-statement} of \tcode{main},
the effect is equivalent to a \tcode{return} with operand \tcode{0}
```

Reply:
Ah, ok! The classic C++ inconsistencies: "`main()` it's a function that must return an `int`, but if you don't return anything, nevermind, I'll do it for you." My suggestion is always to return to avoid such an inconsistency. I won't insist, though.